File: Serene1/Serene1/Serene1.Script/Imports/ServerImports/BasicSamples.LookupFilterByMultipleForm.cs
﻿using jQueryApi;
using Serenity;
using Serenity.ComponentModel;
using System;
using System.Collections;
using System.Collections.Generic;
using System.ComponentModel;
using System.Runtime.CompilerServices;

namespace Serene1.BasicSamples
{
    public partial class LookupFilterByMultipleForm : PrefixedContext
    {
        [InlineConstant]
        public const string FormKey = "BasicSamples.LookupFilterByMultiple";

        public LookupFilterByMultipleForm(string idPrefix) : base(idPrefix) { }

        public StringEditor ProductName { [InlineCode("{this}.w('ProductName', Serenity.StringEditor)")] get; private set; }
        public ImageUploadEditor ProductImage { [InlineCode("{this}.w('ProductImage', Serenity.ImageUploadEditor)")] get; private set; }
        public BooleanEditor Discontinued { [InlineCode("{this}.w('Discontinued', Serenity.BooleanEditor)")] get; private set; }
        public LookupEditor SupplierID { [InlineCode("{this}.w('SupplierID', Serenity.LookupEditor)")] get; private set; }
        public ProduceSeafoodCategoryEditor CategoryID { [InlineCode("{this}.w('CategoryID', Serene1.BasicSamples.ProduceSeafoodCategoryEditor)")] get; private set; }
        public StringEditor QuantityPerUnit { [InlineCode("{this}.w('QuantityPerUnit', Serenity.StringEditor)")] get; private set; }
        public DecimalEditor UnitPrice { [InlineCode("{this}.w('UnitPrice', Serenity.DecimalEditor)")] get; private set; }
        public IntegerEditor UnitsInStock { [InlineCode("{this}.w('UnitsInStock', Serenity.IntegerEditor)")] get; private set; }
        public IntegerEditor UnitsOnOrder { [InlineCode("{this}.w('UnitsOnOrder', Serenity.IntegerEditor)")] get; private set; }
        public IntegerEditor ReorderLevel { [InlineCode("{this}.w('ReorderLevel', Serenity.IntegerEditor)")] get; private set; }
    }
}

File: Serene1/Serene1/Serene1.Web/Modules/Common/Imports/ServerTypings/BasicSamples.LookupFilterByMultipleForm.ts
﻿namespace Serene1.BasicSamples {
    export class LookupFilterByMultipleForm extends Serenity.PrefixedContext {
        static formKey = 'BasicSamples.LookupFilterByMultiple';

    }

    export interface LookupFilterByMultipleForm {
        ProductName(): Serenity.StringEditor;
        ProductImage(): Serenity.ImageUploadEditor;
        Discontinued(): Serenity.BooleanEditor;
        SupplierID(): Serenity.LookupEditor;
        CategoryID(): ProduceSeafoodCategoryEditor;
        QuantityPerUnit(): Serenity.StringEditor;
        UnitPrice(): Serenity.DecimalEditor;
        UnitsInStock(): Serenity.IntegerEditor;
        UnitsOnOrder(): Serenity.IntegerEditor;
        ReorderLevel(): Serenity.IntegerEditor;
    }

    [['ProductName', Serenity.StringEditor], ['ProductImage', Serenity.ImageUploadEditor], ['Discontinued', Serenity.BooleanEditor], ['SupplierID', Serenity.LookupEditor], ['CategoryID', ProduceSeafoodCategoryEditor], ['QuantityPerUnit', Serenity.StringEditor], ['UnitPrice', Serenity.DecimalEditor], ['UnitsInStock', Serenity.IntegerEditor], ['UnitsOnOrder', Serenity.IntegerEditor], ['ReorderLevel', Serenity.IntegerEditor]].forEach(x => LookupFilterByMultipleForm.prototype[<string>x[0]] = function() { return this.w(x[0], x[1]); });
}

